request.go

```diff
@@ -153,8 +153,12 @@ func (s *Server) handleConnect(conn conn, req *Request) error {
 	}
 
 	// Attempt to connect
-	addr := net.TCPAddr{IP: req.realDestAddr.IP, Port: req.realDestAddr.Port}
-	target, err := net.DialTCP("tcp", nil, &addr)
+	addr := fmt.Sprintf("%v:%d", req.realDestAddr.IP, req.realDestAddr.Port)
```

Comment:
I would leave the existing logic and just short cut it if a dialer is given.
Something like:
```
var target net.Conn
var err error
if dial != nil {
   target, err = dial("tcp", addr)
} else {
   addr = :=...
   ...
}
```

Otherwise I think fmt.Sprintf will not properly handle IPv6.